### PR TITLE
Improve unit test error messages and allow empty testthat folders

### PR DESCRIPTION
**Pull request by TimKDJ**
I don't want unit tests to fail when there are no tests present (for example in the jaspModuleTemplate), so I reduced that to just a normal message.
Also some messages were not great, so I tried to make them a bit more clear. Most notably the table diff:

Before:
old-table-unit-test-diff

After:
new-table-unit-test-diff

### CODE REVIEW

**Comment by vandenman** [comment]
is this maybe a time to also look at [testthat 3.0.0](https://cran.r-project.org/web/packages/testthat/vignettes/third-edition.html)? In particular, the [snapshot](https://cran.r-project.org/web/packages/testthat/vignettes/snapshotting.html) functionality looks neat!

**Comment by TimKDJ** [comment]
I definitely want us to move to testthat 3.0, but I forsee it being a bit of work.
Considering I'll need to do some conversions to the tests when we don't need `analysisOption()` anymore (post syntax), I think that'll be a good moment to do it.

**Comment by TimKDJ** [comment]
Also:
- added support for modules that don't use `master` as a default branch (e.g., https://github.com/jasp-stats/jaspCochraneMetaAnalyses)
- fixed weird data errors in unit tests that caused problems in the [abtest](https://github.com/jasp-stats/jaspFrequencies/blob/ff9548d81f4e9bc7bcfbb071621b45f5c30bcc00/tests/testthat/test-abtestbayesian.R#L3-L12) and are also the reason why the PR is failing for [Jill](https://github.com/jasp-stats/jaspEquivalenceTTests/pull/8). For whatever reason calling `remotes::install_local()` causes the next analysis run to produce different results then when this function does not need to be called (i.e., subsequent calls when no code changed in between). By replacing it with `install.packages()` the problem disappears. I've looked into it a bit but could not exactly figure out exactly why this is.

File: R/run.R [hunk 1]
```diff
@@ -151,30 +148,38 @@ initAnalysisRuntime <- function(dataset, makeTests, ...) {
...
-    newMd5Sums <- tools::md5sum(files)
-    if (length(md5Sums) == 0 || !module %in% names(md5Sums) || !all(newMd5Sums %in% md5Sums[[module]])) {
-      if (!msgPrinted) {
-        message("Installing module(s) from source")
-        msgPrinted <- TRUE
+    newMd5Sums <- tools::md5sum(srcFiles)
+    if (length(md5Sums) == 0 || !modulePath %in% names(md5Sums) || !all(newMd5Sums %in% md5Sums[[modulePath]])) {
+      modulePkg <- devtools::as.package(modulePath)$package
+      if (modulePkg %in% loadedNamespaces())
+        devtools::unload(modulePkg)
```
Comment (vandenman) [inline R/run.R:170]: As an example of the comment above, the definition in devtools of this is `pkgload::unload`.

**Review by vandenman — CHANGES_REQUESTED** [review]
looks good to me, one minor comment about header files and some food for thought about devtools

**Comment by TimKDJ** [comment]
I implemented your comments

**Pull request merged**